### PR TITLE
Enhancement/i18n thread safe

### PR DESCRIPTION
**Pull request by brunoocasali**
# Context
This code makes any request free of locale leaking problems.

According to the rails guides: https://guides.rubyonrails.org/i18n.html#managing-the-locale-across-requests
Using `I18n.locale` is not a recommended way to set current locale for a request, using the `with_locale` is the best option!

Btw, searching more about the `http_accept_language` I've found this https://github.com/iain/http_accept_language/issues/68

# Summary of Changes

- Remove `I18n.locale`
- Create shared_examples to test shared behaviour between specs

# Checklist

- [x] Added Unit Tests
- [x] CI Passes
- [ ] Deploys to Heroku on test Correctly (Maintainers will handle)
- [ ] Added Documentation (Service and Code when required)

### CODE REVIEW

**Comment by brunoocasali** [comment]
This PR is related to https://github.com/RefugeRestrooms/refugerestrooms/pull/641 :)

**Comment by DeeDeeG** [comment]
cc @mi-wood again, or @tkwidmer. This seems like a good PR, making sure our internationalization (I18n) is ready for multi-threaded Puma. It even adds tests!

I'm personally not very good with Ruby, so I don't think my review would be super meaningful here. If anyone good with Ruby can give this a second pair of eyes, that would be great. Or at least give it the thumbs up so we can approve it, so the author gets credit for this PR toward their `Hacktoberfest` "5 good pull requests" goals.

**Comment by brunoocasali** [comment]
@DeeDeeG @tkwidmer rebased with latest version of develop branch, can I do something to help this get merged? :)

**Comment by DeeDeeG** [comment]
@brunoocasali thank you for rebasing!

I have simply been backed up with the number of pull requests we got this month, plus other life things.

(I'm trying to work through the remaining pull requests. The fewer of them that are left, the easier reviewing each one becomes, due to not having to worry which one will have merge conflicts, or which one will cause the next one to be harder to test.)

I am going to ensure that all the helpful Pull Requests we have received are either "Approved" or merged by the end of the month. I believe all of your Pull Requests are looking good, and we will be approving and eventually merging them. Thanks for sticking with this as I work through the backlog.

File: spec/controllers/pages_controller_spec.rb
Comment (DeeDeeG) [inline spec/controllers/pages_controller_spec.rb:3]: Hi, @brunoocasali, there is an extra space here. Rubocop is flagging this.

```suggestion
describe PagesController, type: :controller do
```

**Comment by DeeDeeG** [comment]
I do recall the documentation for I18n saying this is the correct way. If there are no issues in manual testing, then I am inclined to merge this.

**Comment by DeeDeeG** [comment]
Okay, this works.

I tested by going to about:preferences in Firefox, updating the "preferred language for displaying pages", then visiting links/refreshing the page at https://refuge-deedeeg-test.herokuapp.com. My "preferred language" was always respected accurately.

(As I understand it, this will only really make a difference if/when we enable multiple workers for Puma some time in the future. Note to other maintainers/readers: see #641 for discussion about multiple workers in Puma. May or may not be merged soon, depending on how review/discussion goes.)

I then compared page load times between:
- https://refuge-deedeeg-test.herokuapp.com (this branch)
- https://staging.refugerestrooms.org (`develop` branch)

I didn't notice a significant difference; either one would be faster or slower seemingly at random. So this pull request does not appear to have any adverse impact on performance.

I'll watch this change on the staging instance, after merging, and hopefully it will continue to behave well as intended in the coming days. In my opinion, this is ready to merge.

File: spec/support/shared_examples/localized_request.rb
Comment (DeeDeeG) [inline spec/support/shared_examples/localized_request.rb:9]: @brunoocasali would it be very difficult to also test loading a page with another locale? So far our test quite runs strictly against the site in English. I would love to have some test for a locale other than English, even if it was a very basic or simple test.

If you could do that, it would be much appreciated. If it's a lot of work, let me know, and I'll still accept this pull request without it.

Thank you, and best regards.

\- DeeDeeg

**Review by DeeDeeG — APPROVED** [review]

File: spec/support/shared_examples/localized_request.rb
Comment (brunoocasali) [reply spec/support/shared_examples/localized_request.rb:9]: Hi @DeeDeeG, thanks for your review! 

Don't you think that way we could start testing some code that are already tested by the framework?

The test in this PR just guarantees that every request will contain some kind of language definition. 
When we add a multi-language switcher like a querystring called "lang" for example we could add another spec here, to check.

if the url contains the "lang" querystring all the messages will be translated properly.

**Comment by brunoocasali** [comment]
> I didn't notice a significant difference; either one would be faster or slower seemingly at random. So this pull request does not appear to have any adverse impact on performance

Yeah, this change will not improve the performance or impact in a negative way either. This just makes sure that there is no way someone will request a page in "pt-BR" and get the answer in "fr" because someone requested nanosecods before :)

File: spec/support/shared_examples/localized_request.rb
Comment (DeeDeeG) [reply spec/support/shared_examples/localized_request.rb:9]: It should be fairly easy to set the locale within a spec's code.

But I can see how that would be unrelated to this pull request. Perhaps when I re-add the language switcher feature, I will `@mention` you to see if you might be interested in adding some tests. No obligation, but certainly it would be appreciated if you wanted to help with that.

Thanks for this, I'm going to merge now.

Comment (brunoocasali) [reply spec/support/shared_examples/localized_request.rb:9]: @DeeDeeG of course! just mention me and I see what can I do ;)

**Pull request merged**
